nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java

```diff
@@ -537,6 +538,26 @@ protected void manageRelatedTextOfTopLevelDocument(CoreSession session, Comment
         }
 
         topLevelDoc.setPropertyValue(RELATED_TEXT_RESOURCES, (Serializable) resources);
+
+        // No need to notify the event when we manage the related text of the top level document
+        topLevelDoc.putContextData(DISABLE_NOTIFICATION_SERVICE, TRUE);
         session.saveDocument(topLevelDoc);
     }
+
+    @Override
+    public DocumentRef getCommentedDocument(CoreSession session, DocumentModel commentDocumentModel) {
+        // Case when commentDocumentModel is already the document being commented
+        DocumentModel commentedDocModel = commentDocumentModel;
+
+        // Case when commentDocumentModel is a comment (can be the first comment or any reply)
+        if (commentDocumentModel.hasSchema(COMMENT_SCHEMA)) {
+            commentedDocModel = session.getDocument(commentDocumentModel.getParentRef());
+        }
+
```

Comment:
To stay consistent with the previous comments I would add
// Case when commentDocumentModel is a first level comment